src/MainWindow.vala

```diff
@@ -152,29 +154,33 @@ public class Sideload.MainWindow : Gtk.ApplicationWindow {
             return;
         }
 
-        if (content_type == REF_CONTENT_TYPE) {
-            flatpak_file = new FlatpakRefFile (file);
-        } else if (content_type == BUNDLE_CONTENT_TYPE) {
-            flatpak_file = new FlatpakBundleFile (file);
-        } else if (content_type == FLATPAK_HTTPS_CONTENT_TYPE) {
-            flatpak_file = new FlatpakRefFile (file);
+        switch (content_type) {
+            case REF_CONTENT_TYPE:
+                flatpak_file = new FlatpakRefFile (file);
+                progress_view = new ProgressView (REF_INSTALL);
+                break;
+            case BUNDLE_CONTENT_TYPE:
+                flatpak_file = new FlatpakBundleFile (file);
+                progress_view = new ProgressView (BUNDLE_INSTALL);
+                break;
+            case FLATPAK_HTTPS_CONTENT_TYPE:
+                flatpak_file = new FlatpakRefFile (file);
+                break;
```

Comment:
`progress_view` isn't initialized in case of this route and might cause null access.

Reply:
Oops, reverted that